server/src/domain/user/user.service.ts

```diff
@@ -129,12 +129,9 @@ export class UserService {
 
   async handleUserDeleteCheck() {
     const users = await this.userRepository.getDeletedUsers();
-    for (const user of users) {
-      if (this.isReadyForDeletion(user)) {
-        await this.jobRepository.queue({ name: JobName.USER_DELETION, data: { id: user.id } });
-      }
-    }
-
+    await this.jobRepository.queueAll(
+      users.filter(this.isReadyForDeletion).map((user) => ({ name: JobName.USER_DELETION, data: { id: user.id } })),
```

Comment:
I'd optimize it to:
```suggestion
      users.flatMap((user) => this.isReadyForDeletion(user) ? [{ name: JobName.USER_DELETION, data: { id: user.id } }] : [])),
```
this removes one iteration through the array

Reply:
Thanks!